src/scripts/main.js

```diff
@@ -1,5 +1,29 @@
 'use strict';
 
 document.addEventListener('click', (e) => {
-  // write code here
+  if (e.target.classList.contains('wall')) {
+    const x = e.clientX;
+    const y = e.clientY;
+    const field = e.target.getBoundingClientRect();
+    const spider = document.querySelector('.spider');
+
+    const borderX = (e.target.offsetWidth - e.target.clientWidth) / 2;
+    const borderY = (e.target.offsetHeight - e.target.clientHeight) / 2;
+
+    const maxLeft = Math.max(0, field.width - spider.offsetWidth - borderX * 2);
+
+    const maxTop = Math.max(
+      0,
+      field.height - spider.offsetHeight - borderY * 2,
+    );
+
+    let spiderLeft = x - field.left - spider.offsetWidth / 2 - borderX;
+    let spiderTop = y - field.top - spider.offsetHeight / 2 - borderY;
+
+    spiderLeft = Math.min(maxLeft, spiderLeft);
+    spiderTop = Math.min(maxTop, spiderTop);
```

Comment:
You are correctly limiting the maximum values for 'spiderLeft' and 'spiderTop', but you should also limit the minimum values. The spider can go outside the wall to the left and top because you don't have a check for the minimum values. You can use 'Math.max' to ensure that 'spiderLeft' and 'spiderTop' are never less than 0.